synapse/storage/databases/main/signatures.py

```diff
@@ -26,7 +26,7 @@
 
 class SignatureWorkerStore(EventsWorkerStore):
     @cached()
-    def get_event_reference_hash(self, event_id: str) -> Dict[str, Dict[str, bytes]]:
+    def get_event_reference_hash(self, event_id: str) -> Mapping[str, bytes]:
```

Comment:
Was this wrong to have the extra level of `Dict[str, ...]` nesting before? If so, I wonder why we didn't spot it until now.

Reply:
I suspect we should fix this separately if it was completely wrong?

Reply:
The reasoning was that if the `@cached` method returns a `T`, the `@cachedList` method (below) must return a `Dict / Mapping[Key, T]`, and we infer `T` to be a `Dict / Mapping[str, bytes]`.

We never call the non-list method so this error no impact.